src/main/java/uk/gov/hmcts/divorce/systemupdate/schedule/SystemProgressHeldCasesTask.java

```diff
@@ -70,66 +65,45 @@ public void run() {
-
-                            // trigger notification to applicant's solicitor
-                            triggerEmailNotification(caseData, caseDetails.getId());
-                        }
-                    }
-                } catch (final CcdManagementException e) {
-                    log.error("Submit event failed for case id: {}, continuing to next case", caseDetails.getId());
-                } catch (final IllegalArgumentException e) {
-                    log.error("Deserialization failed for case id: {}, continuing to next case", caseDetails.getId());
-                }
-            }
+            ccdSearchService
+                .searchForAllCasesWithQuery(Holding, query, user, serviceAuth)
+                .forEach(caseDetails -> submitEvent(caseDetails, user, serviceAuth));
 
-            log.info("Awaiting conditional order scheduled task complete.");
+            log.info("SystemProgressHeldCasesTask scheduled task complete.");
         } catch (final CcdSearchCaseException e) {
-            log.error("Awaiting conditional order schedule task stopped after search error", e);
+            log.error("SystemProgressHeldCasesTask schedule task stopped after search error", e);
         } catch (final CcdConflictException e) {
-            log.info("Awaiting conditional order schedule task stopping "
-                + "due to conflict with another running awaiting conditional order task"
+            log.info("SystemProgressHeldCasesTask schedule task stopping due to conflict with another running task"
             );
         }
     }
 
-    private void triggerEmailNotification(CaseData caseData, Long caseId) {
-        boolean applicant1SolicitorRepresented = caseData.getApplicant1().isRepresented();
-
-        if (applicant1SolicitorRepresented) {
-            log.info("For case id {} applicant is represented by solicitor hence sending conditional order notification email", caseId);
-            conditionalOrderNotification.send(caseData, caseId);
-        } else {
-            log.info(
-                "For case id {} applicant is not represented by solicitor hence not sending conditional order notification email",
-                caseId
-            );
+    private void submitEvent(CaseDetails caseDetails, User user, String serviceAuth) {
+        try {
+            final CaseData caseData = objectMapper.convertValue(caseDetails.getData(), CaseData.class);
+
+            LocalDate dateOfIssue = caseData.getApplication().getIssueDate();
+            log.info("issueDate from caseDataMap {}", dateOfIssue);
+
+            if (dateOfIssue == null) {
+                log.error("Ignoring case id {} with created on {} and modified on {}, as issue date is null",
+                    caseDetails.getId(),
+                    caseDetails.getCreatedDate(),
+                    caseDetails.getLastModified()
+                );
+            } else {
+                if (holdingPeriodService.isHoldingPeriodFinished(dateOfIssue)) {
```

Comment:
why is this needed? shouldn't the ES query do this?

Reply:
haven't changed this, but you're right. I'll remove it